### PR TITLE
Update ForceScheduler example to suit new config

### PR DESCRIPTION
**Pull request by djmitche**
Bug [3416](http://trac.buildbot.net/ticket/3416)

### CODE REVIEW

**Comment by djmitche** [comment]
@tardyp - any thoguhts why this didn't work with b6?

**Comment by tardyp** [comment]
mmh, did I remove the legacy api (without codebase) ? possibly, but I dont remember it

File: master/docs/manual/cfg-schedulers.rst [hunk 1]
```diff
@@ -811,22 +811,27 @@ What you need in your config file is something like::
                  name="force",
                  builderNames=["my-builder"],
 
-                 # will generate a combo box
-                 branch=util.ChoiceStringParameter(name="branch",
-                                                   choices=["main","devel"],
-                                                   default="main"),
+                 codebases = [ util.CodebaseParameter(
```
Comment (rutsky) [inline master/docs/manual/cfg-schedulers.rst]: I think there (and below) shouldn't be spaces around "=" and after "[" according to PEP8

File: master/docs/manual/cfg-schedulers.rst [hunk 2]
```diff
@@ -811,22 +811,27 @@ What you need in your config file is something like::
...
+                     "",
+                     name = "Main repository",
+                     # will generate a combo box
+                     branch=util.ChoiceStringParameter(
+                         name="branch",
+                         choices=["master", "hest", primaryBranch],
+                         default="master"),
+
+                         # will generate nothing in the form, but revision, repository,
+                         # and project are needed by buildbot selfcheduling system so we
```
Comment (rutsky) [inline master/docs/manual/cfg-schedulers.rst]: I think comment is garbled here, take a look at previous version if this comment.

**Comment by djmitche** [comment]
good points - updated

File: master/docs/manual/cfg-schedulers.rst [hunk 3]
```diff
@@ -808,39 +808,45 @@ What you need in your config file is something like::
...
-                                          label="force a make clean",
-                                          default=False),
-                    util.StringParameter(name="pull_url",
-                                         label="optionally give a public Git pull url:<br>",
-                                         default="", size=80)
-                 ])
+        name="force",
+        builderNames=["my-builder"],
+
+        codebases=[util.CodebaseParameter(
```
Comment (sa2ajj) [inline master/docs/manual/cfg-schedulers.rst]: I'd consider:

```
codebases=[
    util.CodebaseParameter(
...
]
```

**Comment by sa2ajj** [comment]
Otherwise, LGTM

**Comment by djmitche** [comment]
Good point - @rutsky, this yours to land or hold off on.

File: master/docs/manual/cfg-schedulers.rst [hunk 4]
```diff
@@ -808,39 +808,47 @@ What you need in your config file is something like::
...
+                    default="master"),
+
+                # will generate nothing in the form, but revision, repository,
+                # and project are needed by buildbot scheduling system so we
+                # need to pass a value ("")
+                revision=util.FixedParameter(name="revision", default=""),
+                repository=util.FixedParameter(name="repository", default=""),
+                project=util.FixedParameter(name="project", default=""),
+            ),
+        ]
```
Comment (rutsky) [inline master/docs/manual/cfg-schedulers.rst]: @djmitche comma is missing here

**Comment by rutsky** [comment]
How can these changes be tested in web UI?
Looks like currently Force Scheduler doesn't show anything in the UI: http://trac.buildbot.net/ticket/3422

File: master/docs/manual/cfg-schedulers.rst (hunk 4 above)
Comment (djmitche) [inline master/docs/manual/cfg-schedulers.rst]: Oh, yes, I was thinking that was an assignment.  Writing patches in airports sucks :)

**Comment by rutsky** [comment]
Please respond to my question about testing these changes.
Or should they be merged without actual checking that that config works in master (it doesn't).

**Comment by djmitche** [comment]
@tardyp will need to speak to the UI testing support.

**Comment by Arytas** [comment]
a way to test graphical stuff in the UI could be selenium

**Pull request merged**